app/models/dialog_field_association_validator.rb

```diff
@@ -1,28 +1,11 @@
 class DialogFieldAssociationValidator
-  def circular_references(associations)
-    return false if associations.empty?
-    association_path_list = initial_paths(associations)
-    association_path_list.each do |path|
-      fieldname_being_triggered = path.last
-      next if associations[fieldname_being_triggered].blank?
-      circular_references = walk_value_path(fieldname_being_triggered, associations, path)
-      return circular_references if circular_references.present?
-    end
-    false
-  end
-
-  private
-
-  def initial_paths(associations)
-    associations.flat_map { |key, values| values.map { |value| [key, value] } }
-  end
-
-  def walk_value_path(fieldname_being_triggered, associations, path)
-    while associations[fieldname_being_triggered].present?
-      return [fieldname_being_triggered, associations[fieldname_being_triggered].first] if path.include?(associations[fieldname_being_triggered].first)
-      path << associations[fieldname_being_triggered]
-      path.flatten!
-      fieldname_being_triggered = path.last
+  def check_for_circular_references(hash, k, collection = [])
+    raise KeyAlreadyExistsError, "#{k} already exists in #{collection}" if collection.include?(k)
+    collection << k
+    hash[k]&.each do |val|
```

Comment:
Should this raise something else if the key isn't there?  It seems like that would be a problem, right?

Reply:
Don't think it'll ever be an issue. It'll never end in a path that has a circular reference, so I don't know why we would care.

Reply:
If the key isn't there, it doesn't mean the key isn't present in the yaml as a valid field, it just means that field has no associations hooked up to it. It's fine.